src/paho/mqtt/client.py

```diff
@@ -2438,18 +2424,6 @@ def _handle_publish(self):
         if len(topic) == 0:
             return MQTT_ERR_PROTOCOL
 
-        # Handle topics with invalid UTF-8
-        # This replaces an invalid topic with a message and the hex
-        # representation of the topic for logging. When the user attempts to
-        # access message.topic in the callback, an exception will be raised.
-        if sys.version_info[0] >= 3:
-            try:
-                print_topic = topic.decode('utf-8')
-            except UnicodeDecodeError:
-                print_topic = "TOPIC WITH INVALID UTF-8: " + str(topic)
-        else:
-            print_topic = topic
-
```

Comment:
I've pushed change on develop that make use of print_topic to keep change from 47348cf